### DISCUSSION TITLE
[java] Unable to build jar with content

### DISCUSSION THREAD

**Discussion by jdwill** [pmd/pmd#3046] [Q&A] [answered]
**Affects PMD Version: 6.0.1** 

**Description: When following instructions for jar creation from https://pmd.github.io/pmd-5.4.1/usage/running.html, the resulting jar does not contain any class files, only a manifest.**

I am attempting to import the PMD jar into my project and extend it.  However, after following the above instructions and importing the resulting jar file in pmd-dist/target into my local Maven repo.  The jar file does not appear to have any meaningful contents.  When I try to import a class, such as AbstractJavaRule, the package cannot be found.  After examining the jar, it appears the only thing inside it is a manifest.

**Comment by adangel** [comment, marked as answer]
@jdwill Please use the latest documentation: https://pmd.github.io/latest/pmd_userdocs_getting_started.html

However, I think, your use case is not documented. As far as I understood, you are using maven already, that's good: Then all you need to do is - add dependencies for pmd. You can have a look at the [pom.xml file of pmd-dist](https://github.com/pmd/pmd/blob/master/pmd-dist/pom.xml) to see all the available dependencies.
If you want to integrate PMD for analyzing Java code, then all you need is pmd-core and pmd-java:

```
    <dependencies>
        <dependency>
            <groupId>net.sourceforge.pmd</groupId>
            <artifactId>pmd-core</artifactId>
            <version>6.0.1</version>
        </dependency>
        <dependency>
            <groupId>net.sourceforge.pmd</groupId>
            <artifactId>pmd-java</artifactId>
            <version>6.0.1</version>
</dependency>
```

PMD is available from [maven central](https://search.maven.org/#search|ga|1|g%3Anet.sourceforge.pmd), so the dependencies are fetched automatically. No need to put jar files manually into your local maven repo.

**Comment by jdwill** [comment]
Thanks, that worked like a charm!
